routemaster/state_machine/api.py

```diff
@@ -219,3 +237,66 @@ def delete_label(app: App, label: LabelRef) -> None:
             old_state=current_state.name,
             new_state=None,
         ))
+
+
+EventProcessor = Callable[[App, State, StateMachine, LabelRef, Any], bool]
+
+
+def process_cron(
+    process: EventProcessor,
+    get_labels: Callable[[StateMachine, State, Any], Iterable[str]],
+    app: App,
+    state_machine: StateMachine,
+    state: State,
+    should_terminate: IsExitingCheck,
+):
+    """
+    Cron event entrypoint.
+    """
+    with app.db.begin() as conn:
+        relevant_labels = get_labels(state_machine, state, conn)
+
+    for label_name in relevant_labels:
+        if should_terminate():
+            break
+
+        with suppress_exceptions(logger):
+            label = LabelRef(name=label_name, state_machine=state_machine.name)
+            could_progress = False
+
+            with app.db.begin() as conn:
+                lock_label(label, conn)
+                current_state = get_current_state(label, state_machine, conn)
+
+                if current_state != state:
+                    continue
+
+                could_progress = process(
+                    app,
+                    state,
+                    state_machine,
+                    label,
+                    conn,
+                )
+
+            if could_progress:
```

Comment:
Even if we don't pass `should_terminate` into `process_transitions`, I wonder whether we should call it at this `if`?

Reply:
Good call 467697fd62b2ddfa92bf454186d0daef5d532314